### PR TITLE
aggregator IR rework

### PR DESCRIPTION
**Pull request by catoverdrive**
This subsumes #4540, which I'll close once I think I can get the rest of this finished tonight.

Currently I've got... something that passes tests in Scala, although I haven't gone through and cleaned up the code for the SeqOps and other CodeAggregator stuff yet. Need to go through and push the changes into python but I thought I'd open this as a status update.

@cseed

### CODE REVIEW

**Comment by tpoterba** [comment]
you have a full day! Though Konrad's talk at 12:00 EST may push some people to the website.

**Comment by cseed** [comment]
High level and the tests look great.  I'll try to take a closer look tonight, but I'm basically ready to approve once the tests pass.

I thought about throwing an error on hl.agg.filter if there isn't an aggregator inside, and I think I agree with you now.  At least, if you use an hl.agg.filter, etc. inside an aggregation, like hl.agg.sum(hl.agg.filter(...)), that should given error rather than doing nothing.  This is important given that this was the old syntax.

**Comment by catoverdrive** [comment]
Cool. There's some things that I want to rework, namely the split logic for `Group()` (either splitting it into a separate IR node as you suggested or reworking some of the CodeAggregator stuff so that the logic becomes the same), but I think that doesn't affect the workingness of this code so I was going to leave that part as-is for now.

File: hail/src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala [hunk 1]
```diff
@@ -284,25 +284,25 @@ class AggregatorsSuite {
...
-      ApplyAggOp(If(
-        ApplyComparisonOp(NEQ(TFloat64()), Ref("a", TFloat64()), F64(10.0)),
-        SeqOp(I32(0), FastSeq(ApplyBinaryPrimOp(Multiply(), Ref("a", TFloat64()), Ref("b", TFloat64()))),
-          aggSig),
-        Begin(FastSeq())),
-        FastSeq(), None, aggSig),
-      (FastIndexedSeq(Row(1.0, 10.0), Row(10.0, 10.0), Row(null, 10.0)), TStruct("a" -> TFloat64(), "b" -> TFloat64())),
-      10.0)
-  }
+//  @Test
```
Comment (cseed) [inline hail/src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala]: I think we can delete this because this is no longer possible, right?

**Pull request merged**